### PR TITLE
Removed boost-log and boost-thread dependencies

### PR DESCRIPTION
**Pull request by eidheim**

### CODE REVIEW

**Comment by eidheim** [comment]
@zalox would you mind have a look through the changes? Tested on most platforms.

File: src/notebook.cc [hunk 1]
```diff
@@ -265,14 +256,11 @@ bool Notebook::save_current() {
...
-  JDEBUG("start");
   if (page!=-1) {
     auto view=get_view(page);
     if(view->get_buffer()->get_modified()){
-      if(!save_modified_dialog(page)) {
-        JDEBUG("end false");
+      if(!save_modified_dialog(page))
         return false;
-      }
     }
```
Comment (zalox) [inline src/notebook.cc]: I have reviewed, but not tested. Looks good. Only thing I saw was this double if statement.

**Comment by eidheim** [comment]
What did you do? My branch is saying: "This branch is 1 commit ahead, 1 commit behind cppit:master.". Try to avoid too advanced git commands.

**Comment by eidheim** [comment]
Also if you made changes, I would like to see these in a commit.

**Comment by eidheim** [comment]
I'm fixing this.

**Comment by zalox** [comment]
I did no changes. I used the github squash method.

**Comment by zalox** [comment]
I didn't realise it did a rebase in the background. I'll fix master now.

**Comment by eidheim** [comment]
Thank you:)

**Comment by eidheim** [comment]
Also it says: "eidheim committed with zalox Removed boost-log and boost-thread dependencies (#202)" instead of the correct: "eidheim Removed boost-log and boost-thread dependencies" in the commit log.

**Comment by zalox** [comment]
Now it shoud be correct.

**Comment by eidheim** [comment]
Perfect now, thank you!:)

**Comment by zalox** [comment]
I don't like the "merged pull request from xxx" generated by github. The squash advertised to to everything in one commit, but it did it with a rebase, so the result got confusing. 

If we merge pull requests on the command line the "merge"-commit isn't added and the log looks a lot cleaner in my opinion. The github magic with closing PR's and issues still happen though.

**Comment by eidheim** [comment]
I agree. One can also remove the "merged pull ..." commit by `git reset --hard HEAD~1` maybe? If it shows up as last commit that is.

**Comment by zalox** [comment]
Just to be clear; if there are conflicts a merge-commit can't be avoided :)

**Pull request merged**